### PR TITLE
PLDM: Implementating LG2 logging

### PR DESCRIPTION
**Pull request by riyadixitagra**

### CODE REVIEW

File: common/utils.cpp [hunk 1]
```diff
@@ -140,9 +143,11 @@ uint8_t readHostEID()
 {
     uint8_t eid{};
     std::ifstream eidFile{HOST_EID_PATH};
+
```
Comment (sagisin) [inline common/utils.cpp:146]: remove extra line added

File: host-bmc/dbus/serialize.cpp [hunk 2]
```diff
@@ -140,8 +143,9 @@ void Serialize::reSerialize(const std::vector<uint16_t> types)
     {
         if (savedObjs.contains(type))
         {
-            std::cerr << "Removing objects of type : " << (unsigned)type
-                      << " from the persistent cache\n";
+            error(
```
Comment (sagisin) [inline host-bmc/dbus/serialize.cpp:146]: This should be an info log, it can go in as a separate commit.

**Review by sagisin — APPROVED** [review]

File: oem/ibm/libpldmresponder/inband_code_update.cpp [hunk 3]
```diff
@@ -314,13 +321,40 @@ void CodeUpdate::setVersions()
...
+                // occured
+                // while
+                // assembling
+                // lid
+                // files
+                // or
+                // while
+                // creating
+                // a tar
+                // ball.
```
Comment (sagisin) [inline oem/ibm/libpldmresponder/inband_code_update.cpp:357]: what happened to this block?

**Comment by Pavithrab7** [comment]
Please solve merge conflicts for this PR.

**Review by jaypadath — APPROVED** [review]

**Review by vkaverap — COMMENTED** [review]
there is one line where clang issue is seen , other than that the changes look good to me

File: libpldmresponder/platform.cpp [hunk 4]
```diff
@@ -68,7 +72,7 @@ void Handler::generate(const pldm::utils::DBusHandler& dBusIntf,
 {
     for (const auto& directory : dir)
     {
-        std::cerr << "checking if : " << directory << "exists" << std::endl;
+        error("checking if : {DIR} exists", "DIR", directory.c_str());
```
Comment (ArchanaKakani) [inline libpldmresponder/platform.cpp:75]: should this be info?

File: common/utils.cpp (hunk 1 above)
Comment (riyadixitagra) [reply common/utils.cpp:146]: Done

File: host-bmc/dbus/serialize.cpp (hunk 2 above)
Comment (riyadixitagra) [reply host-bmc/dbus/serialize.cpp:146]: Sure i will make a PR for improvision of logs

File: host-bmc/dbus/pcie_topology.cpp [hunk 5]
```diff
@@ -157,9 +160,9 @@ uint16_t PCIETopology::getEffecterID()
         hostEffecterParser->getPldmPDR());
     if (stateEffecterPDRs.empty())
     {
-        std::cerr
-            << "PCIe Topology: The state set PDR can not be found, entityType = "
-            << entityType << std::endl;
+        error(
+            "PCIe Topology: The state set PDR can not be found, entityType = {ENTITY_TYP}",
+            "ENTITY_TYP", entityType);
```
[the remark this answers is not in the document]
Comment (riyadixitagra) [reply host-bmc/dbus/pcie_topology.cpp:165]: Done

File: common/utils.cpp [hunk 6]
```diff
@@ -330,10 +335,10 @@ void DBusHandler::setDbusProperty(const DBusMapping& dBusMap,
                 dBusMap.objectPath ==
                     "/xyz/openbmc_project/network/hypervisor/eth1/ipv4/addr0")
             {
-                std::cout << " ,service :" << service.c_str()
-                          << " , interface : " << dBusMap.interface.c_str()
-                          << " , path : " << dBusMap.objectPath.c_str()
-                          << std::endl;
+                info(
+                    " ,service :{SERV} , interface : {INTF} , path : {DBUS_OBJ_PATH}",
```
[the remark this answers is not in the document]
Comment (riyadixitagra) [reply common/utils.cpp:339]: Done

File: libpldmresponder/bios_integer_attribute.cpp [hunk 7]
```diff
@@ -33,13 +36,12 @@ BIOSIntegerAttribute::BIOSIntegerAttribute(const Json& entry,
...
     if (rc != PLDM_SUCCESS)
     {
-        std::cerr << "Wrong filed for integer attribute, ATTRIBUTE_NAME="
-                  << attr.c_str() << " ERRMSG=" << errmsg
-                  << " LOWER_BOUND=" << integerInfo.lowerBound
-                  << " UPPER_BOUND=" << integerInfo.upperBound
-                  << " DEFAULT_VALUE=" << integerInfo.defaultValue
-                  << " SCALAR_INCREMENT=" << integerInfo.scalarIncrement
-                  << "\n";
+        error(
```
[the remark this answers is not in the document]
Comment (riyadixitagra) [reply libpldmresponder/bios_integer_attribute.cpp:39]: Done

File: libpldmresponder/bios_integer_attribute.cpp [hunk 8]
```diff
@@ -33,13 +36,12 @@ BIOSIntegerAttribute::BIOSIntegerAttribute(const Json& entry,
...
+            "Wrong filed for integer attribute, ATTRIBUTE_NAME={ATTR_NAME} ERRMSG= {ERR_MSG} LOWER_BOUND={LOW_BOUND} UPPER_BOUND={UPPER_BOUND} DEFAULT_VALUE={DEF_VAL} SCALAR_INCREMENT={SCALAR_INCREMENT}",
```
[the remark this answers is not in the document]
Comment (riyadixitagra) [reply libpldmresponder/bios_integer_attribute.cpp:40]: Done

File: libpldmresponder/fru.cpp [hunk 9]
```diff
@@ -597,9 +601,9 @@ void FruImpl::buildIndividualFRU(const std::string& fruInterface,
     }
     catch (const std::exception& e)
     {
-        std::cout << "Config JSONs missing for the item "
-                  << " in concurrent add path "
-                  << "interface type, interface = " << fruInterface << "\n";
+        info(
```
[the remark this answers is not in the document]
Comment (riyadixitagra) [reply libpldmresponder/fru.cpp:604]: Yes, I will take it up in log improvision PR

File: libpldmresponder/platform.cpp (hunk 4 above)
Comment (riyadixitagra) [reply libpldmresponder/platform.cpp:75]: yes, I will take this up in log improvision PR

File: libpldmresponder/fru.cpp (hunk 9 above)
[the remark this answers is not in the document]
Comment (riyadixitagra) [reply libpldmresponder/fru.cpp:604]: Yes I will take this up in log improvision PR

File: libpldmresponder/bios_string_attribute.cpp [hunk 10]
```diff
@@ -48,12 +52,11 @@ BIOSStringAttribute::BIOSStringAttribute(const Json& entry,
...
     if (rc != PLDM_SUCCESS)
     {
-        std::cerr << "Wrong field for string attribute, ATTRIBUTE_NAME=" << name
-                  << " ERRMSG=" << errmsg
-                  << " MINIMUM_STRING_LENGTH=" << stringInfo.minLength
-                  << " MAXIMUM_STRING_LENGTH=" << stringInfo.maxLength
-                  << " DEFAULT_STRING_LENGTH=" << stringInfo.defLength
-                  << " DEFAULT_STRING=" << stringInfo.defString << "\n";
+        error(
+            "Wrong field for string attribute, ATTRIBUTE_NAME={ATTR_NAME} ERRMSG={ERR_MSG} MINIMUM_STRING_LENGTH={MIN_LEN} MAXIMUM_STRING_LENGTH={MAX_LEN} DEFAULT_STRING_LENGTH={DEF_LEN} DEFAULT_STRING={DEF_STR}",
```
[the remark this answers is not in the document]
Comment (riyadixitagra) [reply libpldmresponder/bios_string_attribute.cpp:56]: done

File: libpldmresponder/fru.cpp [hunk 11]
```diff
@@ -210,9 +213,9 @@ void FruImpl::buildFRUTable()
                 }
                 catch (const std::exception& e)
                 {
-                    std::cout << "Config JSONs missing for the item "
-                                 "interface type, interface = "
-                              << interface.first << "\n";
+                    info(
```
[the remark this answers is not in the document]
Comment (riyadixitagra) [reply libpldmresponder/fru.cpp:216]: Yes, I will take it up in log improvision PR

File: oem/ibm/libpldmresponder/inband_code_update.cpp (hunk 3 above)
Comment (riyadixitagra) [reply oem/ibm/libpldmresponder/inband_code_update.cpp:357]: this is caused by clang, i retried running clang but there is no change to this

Comment (riyadixitagra) [reply oem/ibm/libpldmresponder/inband_code_update.cpp:357]: Done, it was due to clang.

File: libpldmresponder/bios_config.cpp [hunk 12]
```diff
@@ -716,17 +722,21 @@ void BIOSConfig::traceBIOSUpdate(
...
             break;
         }
         case PLDM_BIOS_STRING:
         case PLDM_BIOS_STRING_READ_ONLY:
         {
             auto value =
                 table::attribute_value::decodeStringEntry(attrValueEntry);
-            std::cout << "BIOS:" << attrName << " updated to value: " << value
-                      << ", by BMC:" << std::boolalpha << isBMC << std::endl;
+            auto bmcChk = isBMC ? "true" : "false";
```
[the remark this answers is not in the document]
Comment (riyadixitagra) [reply libpldmresponder/bios_config.cpp:736]: lt will make logs cluttered, a variable is okay.

**Review by vkaverap — APPROVED** [review]

File: libpldmresponder/bios_config.cpp (hunk 12 above)
[the remark this answers is not in the document]
Comment (ArchanaKakani) [reply libpldmresponder/bios_config.cpp:761]: We can avoid this extra variable and to make the log easily readable we can add Key values on separate lines.
This looks better approach provided LG2 supports this
info(
                "BIOS: {ATTR_NAME}, updated to value: {NEW_VAL}, by BMC: {BMC_CHK}",
                "ATTR_NAME", attrName, "NEW_VAL", value, "BMC_CHK", (isBMC ? "true" : "false"));

**Review by ArchanaKakani — APPROVED** [review]
